app/services/declarations/create.rb

```diff
@@ -30,6 +32,15 @@ class Create
     validate :validates_billable_slot_available
     validate :declaration_date_not_in_the_future
 
+    validates :delivery_partner_id, presence: true, if: -> { Feature.declarations_require_delivery_partner? }
+    validates :delivery_partner_id, inclusion: { in: :available_delivery_partners }, if: -> { delivery_partner_id }
+
+    validates :secondary_delivery_partner_id, absence: true, unless: :delivery_partner_id
+    validates :secondary_delivery_partner_id, inclusion: { in: :available_delivery_partners },
+                                              if: -> { secondary_delivery_partner_id }
+
+    validate :delivery_partners_are_not_the_same, if: :delivery_partner_id
+
```

Comment:
Rather than duplicate this validation - it might be nice if we can add a way of copying errors for matching attributes over from the model(s) the service is wrapping?

My validations are on the relationships rather than ids but I think they could equally be moved to validate on the `_id` attributes themselves?